robotintelligence.cpp

```diff
@@ -20,15 +21,22 @@ RobotIntelligence::~RobotIntelligence() {
 
 void RobotIntelligence::run() {
 	initParticles();
+    double lastTime = timeData.getTime();
+    unsigned int lastMeasurementId = laserData.getMeasurementId()-1;
 	while(true) { //TODO
-		std::vector<double> sensorData = readSensors();
-		evalSensors(sensorData);
+        if(lastMeasurementId == laserData.getMeasurementId())
+            continue;
+
+        lastMeasurementId = laserData.getMeasurementId();
+        double timeStep = timeData.getTime()-lastTime;
+        lastTime = timeData.getTime();
+        moveParticles(timeStep);
+		evalSensors();
+
+        estimatePosition();

```

Comment:
estimate position after resampling always with the same number of particles and same weight